Description: Define and use SAGE_SCRIPTS_DIR in more places
 So distros can install them somewhere other than /usr/bin
Author: Tobias Hansen <thansen@debian.org>
Forwarded: TODO

--- a/sage/src/sage/env.py
+++ b/sage/src/sage/env.py
@@ -161,6 +161,7 @@
 # bunch of sage directories and files
 var('SAGE_LOCAL',          os.path.abspath(sys.prefix))
 var('SAGE_ETC',            join(SAGE_LOCAL, 'etc'))
+var('SAGE_SCRIPTS_DIR',    join(SAGE_LOCAL, 'bin'))
 var('SAGE_INC',            join(SAGE_LOCAL, 'include'))
 var('SAGE_SHARE',          join(SAGE_LOCAL, 'share'))
 var('SAGE_DOC',            join(SAGE_SHARE, 'doc', 'sage'))
--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -4,8 +4,8 @@
 # usage: sage_version [-v]
 #   -v    display the full version banner including release date
 sage_version() {
-    if [ -f "$0-version.sh" ]; then
-        . "$0-version.sh"
+    if [ -f "$SAGE_SCRIPTS_DIR/sage-version.sh" ]; then
+        . "$SAGE_SCRIPTS_DIR/sage-version.sh"
     else
         . "$SAGE_ROOT/src/bin/sage-version.sh"
     fi
@@ -1004,13 +1004,13 @@
     shift
     sage_setup
     if [ "$SAGE_DEBUG" = "no" ]; then
-        gdb -x "$0-gdb-commands" \
-            -args python "$0-ipython" "$@" -i
+        gdb -x "$SAGE_SCRIPTS_DIR/sage-gdb-commands" \
+            -args python "$SAGE_SCRIPTS_DIR/sage-ipython" "$@" -i
     else
         sage_dir=$(sage-python -c 'import os, sage; print(os.path.dirname(sage.__file__))')
         cygdb "$sage_dir" "$SAGE_SRC/sage" \
-            -- -x "$0-gdb-commands" \
-            -args python "$0-ipython" "$@" -i
+            -- -x "$SAGE_SCRIPTS_DIR/sage-gdb-commands" \
+            -args python "$SAGE_SCRIPTS_DIR/sage-ipython" "$@" -i
     fi
     exit $?
 fi
--- a/sage/src/bin/sage-env
+++ b/sage/src/bin/sage-env
@@ -303,7 +303,7 @@
     SAGE_ORIG_PATH=$PATH && export SAGE_ORIG_PATH
     SAGE_ORIG_PATH_SET=True && export SAGE_ORIG_PATH_SET
 fi
-export PATH="$SAGE_ROOT/build/bin:$SAGE_SRC/bin:$SAGE_LOCAL/bin:$PATH"
+export PATH="$SAGE_ROOT/build/bin:$SAGE_SRC/bin:$SAGE_LOCAL/bin:$SAGE_SCRIPTS_DIR:$PATH"
 
 # We offer a toolchain option, so if $SAGE_LOCAL/toolchain/toolchain-env exists source it.
 # Since the user might do something crazy we do not do any checks, but hope for the best.
--- a/sage/src/sage/doctest/control.py
+++ b/sage/src/sage/doctest/control.py
@@ -30,7 +30,7 @@
 import types
 import sage.misc.flatten
 from sage.structure.sage_object import SageObject
-from sage.env import DOT_SAGE, SAGE_LIB, SAGE_SRC, SAGE_LOCAL, SAGE_EXTCODE
+from sage.env import DOT_SAGE, SAGE_LIB, SAGE_SRC, SAGE_LOCAL, SAGE_EXTCODE, SAGE_SCRIPTS_DIR
 from sage.misc.temporary_file import tmp_dir
 from cysignals.signals import AlarmInterrupt, init_cysignals
 
@@ -1072,7 +1072,7 @@
             return 2
         opt = self.options
         if opt.gdb:
-            cmd = '''exec gdb -x "%s" --args '''%(os.path.join(SAGE_LOCAL,"bin","sage-gdb-commands"))
+            cmd = '''exec gdb -x "%s" --args '''%(os.path.join(SAGE_SCRIPTS_DIR,"sage-gdb-commands"))
             flags = ""
             if opt.logfile:
                 sage_cmd += " --logfile %s"%(opt.logfile)
